build.sbt

```diff
@@ -49,6 +50,7 @@ lazy val `kyo-settings` = Seq(
     scalaVersion       := scala3Version,
     crossScalaVersions := List(scala3Version),
     scalacOptions ++= scalacOptionTokens(compilerOptions).value,
+    Test / scalacOptions --= scalacOptionTokens(Set(ScalacOptions.warnNonUnitStatement)).value,
```

Comment:
The tests generate a lot of warnings because there are many `assert` calls that don't use the result so I'm disabling the flag for them.